### PR TITLE
fix: Embed: Get correct snippet for Org events

### LINKED ISSUE #11644: make react embed work for orgs
embed generator has no "organisation" prop yet

image

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

This makes the embed snippet now work with org events without any manual change's requirement.

E2E Tests
- Added e2e tests to verify that the `orgSlug` is present in code and bookerUrl for all types of embeds.
- Now non-org event-types also verify their code
- Code Verification is using eslint and node-html-parser to validate that valid JS code is there in snippets. 

Fixes #11644 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
- Need to test codes in following cases
	- Org + Self Hosting
	- Org + Hosted
	- NonOrg + Self Hosting
	- NonOrg + Hosted
See loom - https://www.loom.com/share/4e6696a6da704ae1b2d8146dd02c2a5a

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12380** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/12380?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/12380?utm_source=stack-comment).

File: apps/web/playwright/embed-code-generator.e2e.ts [hunk 1]
```diff
@@ -1,19 +1,311 @@
...
+
+        await expectToContainValidPreviewIframe(page, {
+          embedType: "inline",
+          calLink: decodeURIComponent(embedUrl),
+        });
+      });
+    });
+  });
+
+  test.describe("Organization", () => {
```
Comment (hariombalhara) [inline apps/web/playwright/embed-code-generator.e2e.ts:178]: New tests to verify the code for organization events.

File: apps/web/playwright/fixtures/users.ts [hunk 2]
```diff
@@ -427,7 +427,7 @@ const createUserFixture = (user: UserWithIncludes, page: Page) => {
       }
       return membership;
     },
-    getOrg: async () => {
+    getOrgMembership: async () => {
```
Comment (hariombalhara) [inline apps/web/playwright/fixtures/users.ts:430]: Renamed to better convey what it does.

File: apps/web/playwright/embed-code-generator.e2e.ts [hunk 3]
```diff
@@ -1,19 +1,332 @@
...
+    });
+  });
+});
+
+type EmbedType = "inline" | "floating-popup" | "element-click";
+function chooseEmbedType(page: Page, embedType: EmbedType) {
   page.locator(`[data-testid=${embedType}]`).click();
 }
 
-async function gotToPreviewTab(page: Page) {
```
Comment (hariombalhara) [inline apps/web/playwright/embed-code-generator.e2e.ts:12]: Renamed to `goToPreviewTab` - Typo fix

File: packages/app-store-cli/src/build.ts [hunk 4]
```diff
@@ -104,10 +104,9 @@ function generateFiles() {
    * If a file has index.ts or index.tsx, it can be imported after removing the index.ts* part
    */
   function getModulePath(path: string, moduleName: string) {
-    return (
-      `./${path.replace(/\\/g, "/")}/` +
-      moduleName.replace(/\/index\.ts|\/index\.tsx/, "").replace(/\.tsx$|\.ts$/, "")
-    );
+    return `./${path.replace(/\\/g, "/")}/${moduleName
+      .replace(/\/index\.ts|\/index\.tsx/, "")
+      .replace(/\.tsx$|\.ts$/, "")}`;
```
Comment (hariombalhara) [inline packages/app-store-cli/src/build.ts:109]: Prettier autofix

File: packages/features/embed/lib/EmbedCodes.tsx [hunk 5]
```diff
@@ -53,7 +55,7 @@ export const Codes = {
       return code`
   import { getCalApi } from "@calcom/embed-react";
   import { useEffect } from "react";
-  export default function App() {
+  export default function MyApp() {
```
Comment (hariombalhara) [inline packages/features/embed/lib/EmbedCodes.tsx:58]: Renamed to make it same as `inline` embed code

**Review by hariombalhara — COMMENTED** [review]
Self review done

**Review by zomars — APPROVED** [review]
Great work @hariombalhara   looks good enough to merge

**Pull request merged**